test/mortgage_e2e.ipynb

```diff
@@ -0,0 +1,1123 @@
+    "    perf_df_tmp = gpu_load_performance_csv(perf_file)\n",
+    "    gdf = perf_df_tmp\n",
+    "    everdf = create_ever_features(gdf)\n",
+    "    delinq_merge = create_delinq_features(gdf)\n",
+    "    everdf = join_ever_delinq_features(everdf, delinq_merge)\n",
+    "    del(delinq_merge)\n",
+    "    joined_df = create_joined_df(gdf, everdf)\n",
+    "    testdf = create_12_mon_features(joined_df)\n",
+    "    joined_df = combine_joined_12_mon(joined_df, testdf)\n",
+    "    del(testdf)\n",
+    "    perf_df = final_performance_delinquency(gdf, joined_df)\n",
+    "    del(gdf, joined_df)\n",
+    "    final_gdf = join_perf_acq_gdfs(perf_df, acq_gdf)\n",
+    "    del(perf_df)\n",
+    "    del(acq_gdf)\n",
+    "    final_gdf = last_mile_cleaning(final_gdf)\n",
+    "    return final_gdf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## ETL"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Perform all of ETL with a single call to\n",
+    "```python\n",
+    "process_quarter_gpu(year=year, quarter=quarter, perf_file=file)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "ETL for start_year:2000 and end_year:2007\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "DoneAndNotDoneFutures(done=set(), not_done=set())"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "\n",
+    "# NOTE: The ETL calculates additional features which are then dropped before creating the XGBoost DMatrix.\n",
+    "# This can be optimized to avoid calculating the dropped features.\n",
+    "part_count = 4\n",
+    "\n",
+    "gpu_dfs = []\n",
+    "gpu_time = 0\n",
+    "quarter = 1\n",
+    "year = start_year\n",
+    "count = 0\n",
+    "while year <= end_year:\n",
+    "    for file in glob(os.path.join(perf_data_path + \"/Performance_\" + str(year) + \"Q\" + str(quarter) + \"*\")):\n",
+    "        gpu_dfs.append(process_quarter_gpu(year=year, quarter=quarter, perf_file=file))\n",
+    "        count += 1\n",
+    "    quarter += 1\n",
+    "    if quarter == 5:\n",
+    "        year += 1\n",
+    "        quarter = 1\n",
+    "print(\"ETL for start_year:{} and end_year:{}\\n\".format(start_year,end_year))\n",
+    "wait(gpu_dfs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'tcp://127.0.0.1:33433': None, 'tcp://127.0.0.1:40907': None}"
+      ]
+     },
+     "execution_count": 23,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "client.run(initialize_rmm_no_pool)"
```

Comment:
We would want to do this after the `persist` call since we want all the dask-cudf operations to happen with the pool.